From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 7 Oct 2022 20:59:10 -0400
Subject: tests: Ignore deprecations

Eventually, most of these tests should be dropped.

Origin: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/9721>
Applied-upstream: 3.24.53, commit:ebf2582da6057630ef5e3a09b667246481e005b1
---
 tests/overlayscroll.c        | 2 ++
 tests/testaccel.c            | 2 ++
 tests/testbaseline.c         | 2 ++
 tests/testcellarea.c         | 2 ++
 tests/testcellrenderertext.c | 2 ++
 tests/testcombo.c            | 2 ++
 tests/testcombochange.c      | 2 ++
 tests/testellipsise.c        | 2 ++
 tests/testflowbox.c          | 2 ++
 tests/testfontoptions.c      | 2 ++
 tests/testgmenu.c            | 2 ++
 tests/testgrouping.c         | 1 +
 tests/testgtk.c              | 2 ++
 tests/testiconview-keynav.c  | 2 ++
 tests/testiconview.c         | 2 ++
 tests/testinfobar.c          | 2 ++
 tests/testkineticscrolling.c | 2 ++
 tests/testlist.c             | 2 ++
 tests/testlist3.c            | 2 ++
 tests/testmenubutton.c       | 2 ++
 tests/testpopover.c          | 2 ++
 tests/testscrolledwindow.c   | 1 +
 tests/teststack.c            | 2 ++
 tests/testtooltips.c         | 2 ++
 tests/testtreechanging.c     | 2 ++
 tests/testtreecolumns.c      | 2 ++
 tests/testtreecolumnsizing.c | 2 ++
 tests/testtreednd.c          | 2 ++
 tests/testtreeedit.c         | 2 ++
 tests/testtreeflow.c         | 2 ++
 tests/testtreefocus.c        | 2 ++
 tests/testtreemodel.c        | 2 ++
 tests/testtreepos.c          | 2 ++
 tests/testtreesort.c         | 1 +
 tests/testtreeview.c         | 2 ++
 tests/testverticalcells.c    | 2 ++
 tests/treestoretest.c        | 2 ++
 37 files changed, 71 insertions(+)

diff --git a/tests/overlayscroll.c b/tests/overlayscroll.c
index e432331..d250401 100644
--- a/tests/overlayscroll.c
+++ b/tests/overlayscroll.c
@@ -18,6 +18,8 @@
 #include <gtk/gtk.h>
 #include <string.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static gchar *
 get_content (void)
 {
diff --git a/tests/testaccel.c b/tests/testaccel.c
index ef213c9..4770a12 100644
--- a/tests/testaccel.c
+++ b/tests/testaccel.c
@@ -18,6 +18,8 @@
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static void
 accel_edited_callback (GtkCellRendererText *cell,
                        const char          *path_string,
diff --git a/tests/testbaseline.c b/tests/testbaseline.c
index a9581f7..b834aa4 100644
--- a/tests/testbaseline.c
+++ b/tests/testbaseline.c
@@ -18,6 +18,8 @@
 
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static char *baseline_pos_str[] = {
   "BASELINE_POSITION_TOP",
   "BASELINE_POSITION_CENTER",
diff --git a/tests/testcellarea.c b/tests/testcellarea.c
index 1c2a85a..36a116e 100644
--- a/tests/testcellarea.c
+++ b/tests/testcellarea.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 /*******************************************************
  *                      Simple Test                    *
  *******************************************************/
diff --git a/tests/testcellrenderertext.c b/tests/testcellrenderertext.c
index e57964e..f1285f0 100644
--- a/tests/testcellrenderertext.c
+++ b/tests/testcellrenderertext.c
@@ -21,6 +21,8 @@
 
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 #define COL_BACKGROUND 15
 #define COL_LINE_NUM   16
 #define NUM_COLS       17 /* change this when adding columns */
diff --git a/tests/testcombo.c b/tests/testcombo.c
index 44469bf..a978064 100644
--- a/tests/testcombo.c
+++ b/tests/testcombo.c
@@ -19,6 +19,8 @@
 #include <string.h>
 #include <stdio.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 /**
  * oh yes, this test app surely has a lot of ugly code
  */
diff --git a/tests/testcombochange.c b/tests/testcombochange.c
index fdd4772..8714b2e 100644
--- a/tests/testcombochange.c
+++ b/tests/testcombochange.c
@@ -20,6 +20,8 @@
 #include <gtk/gtk.h>
 #include <stdarg.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 GtkWidget *text_view;
 GtkListStore *model;
 GArray *contents;
diff --git a/tests/testellipsise.c b/tests/testellipsise.c
index 6e78ff2..32cc840 100644
--- a/tests/testellipsise.c
+++ b/tests/testellipsise.c
@@ -24,6 +24,8 @@
 
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static void
 redraw_event_box (GtkWidget *widget)
 {
diff --git a/tests/testflowbox.c b/tests/testflowbox.c
index ad18121..24e094a 100644
--- a/tests/testflowbox.c
+++ b/tests/testflowbox.c
@@ -17,6 +17,8 @@
 
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 enum {
   SIMPLE_ITEMS = 0,
   FOCUS_ITEMS,
diff --git a/tests/testfontoptions.c b/tests/testfontoptions.c
index 644ece9..8e9ba1e 100644
--- a/tests/testfontoptions.c
+++ b/tests/testfontoptions.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static GtkWidget *antialias;
 static GtkWidget *subpixel;
 static GtkWidget *hintstyle;
diff --git a/tests/testgmenu.c b/tests/testgmenu.c
index 246a4ed..a7dc82e 100644
--- a/tests/testgmenu.c
+++ b/tests/testgmenu.c
@@ -21,6 +21,8 @@
 #include <gio/gio.h>
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 /* TODO
  *
  * - Labeled sections
diff --git a/tests/testgrouping.c b/tests/testgrouping.c
index 4b412d0..bf2a5d8 100644
--- a/tests/testgrouping.c
+++ b/tests/testgrouping.c
@@ -1,5 +1,6 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
 
 static GtkTreeModel *
 create_model (void)
diff --git a/tests/testgtk.c b/tests/testgtk.c
index b945699..7913409 100644
--- a/tests/testgtk.c
+++ b/tests/testgtk.c
@@ -47,6 +47,8 @@
 
 #include "test.xpm"
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 gboolean
 file_exists (const char *filename)
 {
diff --git a/tests/testiconview-keynav.c b/tests/testiconview-keynav.c
index 3125363..f506802 100644
--- a/tests/testiconview-keynav.c
+++ b/tests/testiconview-keynav.c
@@ -25,6 +25,8 @@
 
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static GtkTreeModel *
 get_model (void)
 {
diff --git a/tests/testiconview.c b/tests/testiconview.c
index 010109a..25cb52c 100644
--- a/tests/testiconview.c
+++ b/tests/testiconview.c
@@ -19,6 +19,8 @@
 #include <sys/types.h>
 #include <string.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 #define NUMBER_OF_ITEMS   10
 #define SOME_ITEMS       100
 #define MANY_ITEMS     10000
diff --git a/tests/testinfobar.c b/tests/testinfobar.c
index 3762bb1..49233ba 100644
--- a/tests/testinfobar.c
+++ b/tests/testinfobar.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 typedef enum {
   RESPONSE_UNREVEAL,
 } Response;
diff --git a/tests/testkineticscrolling.c b/tests/testkineticscrolling.c
index dec82da..f91ff3b 100644
--- a/tests/testkineticscrolling.c
+++ b/tests/testkineticscrolling.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 enum
 {
   TARGET_GTK_TREE_MODEL_ROW
diff --git a/tests/testlist.c b/tests/testlist.c
index a96722b..c844394 100644
--- a/tests/testlist.c
+++ b/tests/testlist.c
@@ -2,6 +2,8 @@
 
 #include <string.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 typedef struct _Row Row;
 typedef struct _RowClass RowClass;
 
diff --git a/tests/testlist3.c b/tests/testlist3.c
index 9a8d1c6..5b8b0bc 100644
--- a/tests/testlist3.c
+++ b/tests/testlist3.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static GtkTargetEntry entries[] = {
   { "GTK_LIST_BOX_ROW", GTK_TARGET_SAME_APP, 0 }
 };
diff --git a/tests/testmenubutton.c b/tests/testmenubutton.c
index 8aadb81..4b4a3c0 100644
--- a/tests/testmenubutton.c
+++ b/tests/testmenubutton.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 #define INITIAL_HALIGN          GTK_ALIGN_START
 #define INITIAL_VALIGN          GTK_ALIGN_START
 
diff --git a/tests/testpopover.c b/tests/testpopover.c
index addba18..1d07fe5 100644
--- a/tests/testpopover.c
+++ b/tests/testpopover.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static void
 activate (GSimpleAction *action,
           GVariant      *parameter,
diff --git a/tests/testscrolledwindow.c b/tests/testscrolledwindow.c
index 47677d3..e4c2e4a 100644
--- a/tests/testscrolledwindow.c
+++ b/tests/testscrolledwindow.c
@@ -1,5 +1,6 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
 
 static void
 horizontal_policy_changed (GtkComboBox *combo_box,
diff --git a/tests/teststack.c b/tests/teststack.c
index 07890e1..56f8414 100644
--- a/tests/teststack.c
+++ b/tests/teststack.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 GtkWidget *stack;
 GtkWidget *switcher;
 GtkWidget *sidebar;
diff --git a/tests/testtooltips.c b/tests/testtooltips.c
index 853239e..94517db 100644
--- a/tests/testtooltips.c
+++ b/tests/testtooltips.c
@@ -23,6 +23,8 @@
 
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static gboolean
 query_tooltip_cb (GtkWidget  *widget,
 		  gint        x,
diff --git a/tests/testtreechanging.c b/tests/testtreechanging.c
index f974486..00cb368 100644
--- a/tests/testtreechanging.c
+++ b/tests/testtreechanging.c
@@ -18,6 +18,8 @@
 
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 #define MIN_ROWS 50
 #define MAX_ROWS 150
 
diff --git a/tests/testtreecolumns.c b/tests/testtreecolumns.c
index 4c1d87f..0713935 100644
--- a/tests/testtreecolumns.c
+++ b/tests/testtreecolumns.c
@@ -19,6 +19,8 @@
 #include "config.h"
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 /*
  * README README README README README README README README README README
  * README README README README README README README README README README
diff --git a/tests/testtreecolumnsizing.c b/tests/testtreecolumnsizing.c
index 2ba5631..3bfc56b 100644
--- a/tests/testtreecolumnsizing.c
+++ b/tests/testtreecolumnsizing.c
@@ -23,6 +23,8 @@
 #include <gtk/gtk.h>
 #include <string.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 #define NO_EXPAND "No expandable columns"
 #define SINGLE_EXPAND "One expandable column"
 #define MULTI_EXPAND "Multiple expandable columns"
diff --git a/tests/testtreednd.c b/tests/testtreednd.c
index 0a84ec3..270e884 100644
--- a/tests/testtreednd.c
+++ b/tests/testtreednd.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 typedef GtkListStore MyModel;
 typedef GtkListStoreClass MyModelClass;
 
diff --git a/tests/testtreeedit.c b/tests/testtreeedit.c
index ff25735..b0a9373 100644
--- a/tests/testtreeedit.c
+++ b/tests/testtreeedit.c
@@ -19,6 +19,8 @@
 #include "config.h"
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 typedef struct {
   const gchar *string;
   gboolean is_editable;
diff --git a/tests/testtreeflow.c b/tests/testtreeflow.c
index b243bc5..d197306 100644
--- a/tests/testtreeflow.c
+++ b/tests/testtreeflow.c
@@ -19,6 +19,8 @@
 #include "config.h"
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 GtkTreeModel *model = NULL;
 static GRand *grand = NULL;
 GtkTreeSelection *selection = NULL;
diff --git a/tests/testtreefocus.c b/tests/testtreefocus.c
index 024b1bd..3fae7c0 100644
--- a/tests/testtreefocus.c
+++ b/tests/testtreefocus.c
@@ -19,6 +19,8 @@
 #include "config.h"
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 typedef struct _TreeStruct TreeStruct;
 struct _TreeStruct
 {
diff --git a/tests/testtreemodel.c b/tests/testtreemodel.c
index c240b18..a6e8e0a 100644
--- a/tests/testtreemodel.c
+++ b/tests/testtreemodel.c
@@ -25,6 +25,8 @@
 
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static gint repeats = 2;
 static gint max_size = 8;
 
diff --git a/tests/testtreepos.c b/tests/testtreepos.c
index 77cf205..887e1e8 100644
--- a/tests/testtreepos.c
+++ b/tests/testtreepos.c
@@ -1,5 +1,7 @@
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 static gboolean
 clicked_icon (GtkTreeView  *tv,
               gint          x,
diff --git a/tests/testtreesort.c b/tests/testtreesort.c
index c276c6d..b98cbf3 100644
--- a/tests/testtreesort.c
+++ b/tests/testtreesort.c
@@ -19,6 +19,7 @@
 #include "config.h"
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
 
 typedef struct _ListSort ListSort;
 struct _ListSort
diff --git a/tests/testtreeview.c b/tests/testtreeview.c
index 8e00395..7944615 100644
--- a/tests/testtreeview.c
+++ b/tests/testtreeview.c
@@ -20,6 +20,8 @@
 #include <gtk/gtk.h>
 #include <stdlib.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 /* Don't copy this bad example; inline RGB data is always a better
  * idea than inline XPMs.
  */
diff --git a/tests/testverticalcells.c b/tests/testverticalcells.c
index e9e2bcd..38b7c19 100644
--- a/tests/testverticalcells.c
+++ b/tests/testverticalcells.c
@@ -22,6 +22,8 @@
 #include "config.h"
 #include <gtk/gtk.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 typedef struct _TreeEntry TreeEntry;
 
 struct _TreeEntry {
diff --git a/tests/treestoretest.c b/tests/treestoretest.c
index 6730689..b945496 100644
--- a/tests/treestoretest.c
+++ b/tests/treestoretest.c
@@ -20,6 +20,8 @@
 #include <stdlib.h>
 #include <string.h>
 
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
 GtkTreeStore *base_model;
 static gint node_count = 0;
 
